### ISSUE TITLE
nan value formatting failure in the compact mode

### ISSUE THREAD

**Issue by borisxm** [hgrecco/pint#442]
Consider the following code:

```
from pint import UnitRegistry
from math import nan

ureg = UnitRegistry()
Q_ = ureg.Quantity
v = Q_(nan, 'mm')

'{:C}'.format(v)
'{:#}'.format(v)
```

The first 'format' statement properly produces 'nan millimeter' while second fails with:
ValueError: cannot convert float NaN to integer.

**Comment by hgrecco** [comment]
Full traceback:
```
>>> '{:#}'.format(v)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pint/quantity.py", line 137, in __format__
    obj = self.to_compact()
  File "pint/quantity.py", line 349, in to_compact
    power = int(math.floor(math.log10(magnitude) / unit_power / 3)) * 3
ValueError: cannot convert float NaN to integer
```

Same problem as #444

**Comment by hgrecco** [comment]
Should be fixed now due to merge of #443 
Reopen if necessary
